yoyodyne/evaluators.py

```diff
@@ -8,16 +11,55 @@ class Error(Exception):
     pass
 
 
+@dataclass
+class EvalItem:
+    num_correct: int
+    num_predicted: int
+
+    @property
+    def accuracy(self) -> float:
+        """Computes the accuracy."""
+        return self.num_correct / self.num_predicted
+
+    def __add__(self, other_eval: EvalItem) -> EvalItem:
+        """Adds two EvalItems by summing along both attributes.
+
+        Args:
+            other_eval (EvalItem): The other eval item to add to self.
+
+        Returns:
+            EvalItem.
+        """
+        return EvalItem(
+            self.num_correct + other_eval.num_correct,
+            self.num_predicted + other_eval.num_predicted,
+        )
+
+    def __radd__(self, start_val: int) -> EvalItem:
+        """The initial add when calling `sum` on an iterable of EvalItems.
+        A zero values start value is provided.
```

Comment:
It doesn't look like this statement is true.

Reply:
Maybe it is my comment that is confusing, but here is what I thought happens when sum is called on an iterable of `EvalItem`s:

1. (`sum`)[https://docs.python.org/3/library/functions.html#sum] calls __add__ between the first `EvalItem` and the `start` values, which defaults to 0.
2. `__add__` for `int` raises NotImplementedError because `int` does not define `__add__` with `EvalItem`, and sum first calls `__add__` for the left type. 
3. Some python magic happens and it then checks for the "reverse add", that is, `__radd__` on the right type. I have defined it above so that it takes the default `0` and adds it to numerator and denominator, behaving very similarly to adding two `EvalItems`.

Note that removing this `__radd__` implementation causes:

```
TypeError: unsupported operand type(s) for +: 'int' and 'EvalItem'
```

to be raised.

Reply:
> Maybe it is my comment that is confusing, but here is what I thought happens when sum is called on an iterable of `EvalItem`s:
> 
> 1. [sum](https://docs.python.org/3/library/functions.html#sum) calls **add** between the first `EvalItem` and the `start` value, which defaults to 0.
> 2. `__add__` for `int` raises NotImplementedError because `int` does not define `__add__` with `EvalItem`, and sum first calls `__add__` for the left type.

Minor correction is that I think it `return NotImplemented` which is the new way to do this.

> 3. Some python magic happens and it then checks for the "reverse add", that is, `__radd__` on the right type. I have defined it above so that it takes the default `0` and adds it to numerator and denominator, behaving very similarly to adding two `EvalItems`.

Yeah that's all right. I just didn't find the comment helpful, since this is true of the `sum`/`radd` interaction in general?

Reply:
Yeah good point. The comment is confusing. Does the updated version help at all? Since this behavior is built-in to python, I guess we do not need to document too much of it in the method implementation.